archo.cpp

```diff
@@ -2,9 +2,239 @@
 #include "common/json.h"
 #include "archo.h"
 #include "signing.h"
+#include <sstream>
 
 size_t execSegLimit = 0;
 
+static inline void put(std::streambuf& stream, const void* data, size_t size) {
+    stream.sputn(static_cast<const char*>(data), size);
+}
+
+struct EntitlementValue
+{
+    enum Type
+    {
+        Boolean = 0x01,
+        String = 0x0c,
+    };
+
+    Type type;
+    uint8_t length; // Excludes .type and .length
+    std::string value;
+
+    EntitlementValue(Type type, std::string value) : type(type), length(value.length()), value(value)
+    {
+    }
+
+    uint8_t totalLength()
+    {
+        return this->length + 2;
+    }
+
+    std::string data()
+    {
+        std::stringbuf data;
+
+        put(data, (char*)& this->type, 1);
+        put(data, (char*)& this->length, 1);
```

Comment:
if this->length is larger than 127, we should put the value `x80 | bytesof(this->length)` in one byte which is followed by `bytesof(length)` bytes of `length`